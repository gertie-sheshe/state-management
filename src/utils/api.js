const API_URL = "http://localhost:3000";
const HEADERS = {
  Accept: "application/json",
  "Content-Type": "application/json",
};

export const apiRequest = async (url, options) => {
  const result = await fetch(url, options);
  const data = await result.json();

  return data;
};

export const getUsers = () => {
  return apiRequest(`${API_URL}/users`);
};

export const getTodos = () => {
  return apiRequest(`${API_URL}/todos`);
};

export const getUserTodos = (userId) => {
  return apiRequest(`${API_URL}/users/${userId}/todos`);
};

export const updateTodo = (todoId, data) => {
  return apiRequest(`${API_URL}/todos/${todoId}`, {
    method: "PUT",
    headers: HEADERS,
    body: JSON.stringify(data),
  });
};

export const createTodo = (data, userId) => {
  return apiRequest(`${API_URL}/users/${userId}/todos`, {
    method: "POST",
    headers: HEADERS,
    body: JSON.stringify(data),
  });
};
