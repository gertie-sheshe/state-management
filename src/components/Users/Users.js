import React from "react";
import { useQuery } from "react-query";
import { getUsers } from "../../utils/api";

function Users({ handleSelectChange }) {
  const { data, isLoading, isError } = useQuery("users", getUsers);

  if (isError) {
    return <p>Error: {isError}</p>;
  }

  if (isLoading) {
    return <p>Loading...</p>;
  }
  return (
    <>
      <label htmlFor="users">Choose User: </label>
      <select onChange={handleSelectChange} name="users" id="users">
        <option defaultValue>Select User</option>
        {data?.map((user) => {
          return (
            <option value={user.id} key={user.id}>
              {user.name}
            </option>
          );
        })}
      </select>
    </>
  );
}

export default Users;
