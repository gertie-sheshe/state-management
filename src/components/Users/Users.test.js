import { render } from "../../tests/utils";
import Users from "./Users";

describe("Users", () => {
  test("Matches snapshot", () => {
    const { asFragment } = render(<Users />);
    expect(asFragment()).toMatchSnapshot();
  });
});
