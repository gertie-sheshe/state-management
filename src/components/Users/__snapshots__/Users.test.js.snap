// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users Matches snapshot 1`] = `
<DocumentFragment>
  <p>
    Loading...
  </p>
</DocumentFragment>
`;
